librz/core/cmd/cmd_analysis.c

```diff
@@ -845,32 +830,121 @@ static void cmd_syscall_do(RzCore *core, st64 n, ut64 addr) {
+	core->parser->subrel = rz_config_get_i(core->config, "asm.sub.rel");
+	int ret, i, j, idx;
+	RzAnalysisOp op = { 0 };
+	ut64 addr;
+	int totalsize = 0;
+
+	for (i = idx = 0; idx < len && (!nops || (nops && i < nops)); i++, idx += ret) {
+		addr = core->offset + idx;
+		rz_asm_set_pc(core->rasm, addr);
+		ret = rz_analysis_op(core->analysis, &op, addr, buf + idx, len - idx,
+			RZ_ANALYSIS_OP_MASK_ESIL | RZ_ANALYSIS_OP_MASK_IL | RZ_ANALYSIS_OP_MASK_OPEX | RZ_ANALYSIS_OP_MASK_HINT);
+
+		if (ret < 1) {
+			eprintf("Oops at 0x%08" PFMT64x " (", core->offset + idx);
+			for (i = idx, j = 0; i < core->blocksize && j < 3; i++, j++) {
+				eprintf("%02x ", buf[i]);
+			}
+			eprintf("...)\n");
+			break;
+		}
+		totalsize += op.size;
+		rz_analysis_op_fini(&op);
+	}
+	rz_analysis_op_fini(&op);
+	rz_cons_printf("%d\n", totalsize);
+}
+
+static void core_analysis_bytes_desc(RzCore *core, const ut8 *buf, int len, int nops) {
+	core->parser->subrel = rz_config_get_i(core->config, "asm.sub.rel");
+	int ret, i, idx;
+	RzAsmOp asmop;
+	RzAnalysisOp op = { 0 };
+	ut64 addr;
+
+	for (i = idx = 0; idx < len && (!nops || (nops && i < nops)); i++, idx += ret) {
+		addr = core->offset + idx;
+		rz_asm_set_pc(core->rasm, addr);
+		ret = rz_analysis_op(core->analysis, &op, addr, buf + idx, len - idx,
+			RZ_ANALYSIS_OP_MASK_ESIL | RZ_ANALYSIS_OP_MASK_IL | RZ_ANALYSIS_OP_MASK_OPEX | RZ_ANALYSIS_OP_MASK_HINT);
+		(void)rz_asm_disassemble(core->rasm, &asmop, buf + idx, len - idx);
+
+		char *opname = strdup(rz_asm_op_get_asm(&asmop));
+		if (opname) {
+			rz_str_split(opname, ' ');
+			char *d = rz_asm_describe(core->rasm, opname);
+			if (d && *d) {
+				rz_cons_printf("%s: %s\n", opname, d);
+				free(d);
+			} else {
+				eprintf("Unknown opcode\n");
+			}
+			free(opname);
+		}
+		rz_analysis_op_fini(&op);
+	}
+	rz_analysis_op_fini(&op);
+}
+
+static void core_analysis_bytes_esil(RzCore *core, const ut8 *buf, int len, int nops) {
 	bool use_color = core->print->flags & RZ_PRINT_FLAGS_COLOR;
 	core->parser->subrel = rz_config_get_i(core->config, "asm.sub.rel");
-	int ret, i, j, idx, size;
+	int ret, i, j, idx;
 	const char *color = "";
 	const char *esilstr;
-	const char *opexstr;
-	RzAnalysisHint *hint;
 	RzAnalysisEsil *esil = NULL;
-	RzAsmOp asmop;
 	RzAnalysisOp op = { 0 };
 	ut64 addr;
-	PJ *pj = NULL;
-	int totalsize = 0;
 
 	if (use_color) {
 		color = core->cons->context->pal.label;
 	}
-	if (fmt == 'j') {
-		pj = pj_new();
-		if (!pj) {
-			return;
+	for (i = idx = 0; idx < len && (!nops || (nops && i < nops)); i++, idx += ret) {
+		addr = core->offset + idx;
+		rz_asm_set_pc(core->rasm, addr);
+		ret = rz_analysis_op(core->analysis, &op, addr, buf + idx, len - idx,
+			RZ_ANALYSIS_OP_MASK_ESIL | RZ_ANALYSIS_OP_MASK_IL | RZ_ANALYSIS_OP_MASK_OPEX | RZ_ANALYSIS_OP_MASK_HINT);
+		esilstr = RZ_STRBUF_SAFEGET(&op.esil);
+
+		if (ret < 1) {
+			eprintf("Oops at 0x%08" PFMT64x " (", core->offset + idx);
```

Comment:
Same.

Reply:
Thanks for your review. Changed as you commented.